lib/src/components/data_table/sort.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:angular2/angular2.dart';
+import 'package:skawa_components/src/components/data_table/data_table_column.dart';
+
+enum SortDirection { asc, desc }
+
+class SortModel {
+  final List<SortDirection> allowedDirections;
+
+  SortDirection activeSort;
+
+  SortModel(this.allowedDirections);
+
+  void toggleSort() {
+    if ((allowedDirections ?? const []).isEmpty) {
+      throw new ArgumentError('SortModel does not have any allowed sort directions');
+    }
+    if (activeSort == null) {
+      activeSort = allowedDirections.first;
+    } else {
+      int directionIndex = allowedDirections.indexOf(activeSort);
+      if (directionIndex == allowedDirections.length - 1) {
+        activeSort = null;
+      } else {
+        activeSort = allowedDirections[directionIndex + 1];
+      }
+    }
+  }
+
+  bool get isAscending => activeSort == SortDirection.asc;
+
+  bool get isDescending => activeSort == SortDirection.desc;
+
+  bool get isSorted => activeSort != null;
+}
+
+@Directive(selector: '[sort]')
+class SkawaDataTableSortDirective {

```

Comment:
Wouldn't be easier just define it in the column definition and additionally add the sort fuction to the column?

Reply:
Maybe so. My intention was to implement sort as a (mostly) reusable add-on on top of the data table, and decouple it from the table as much as possible. Consider, for example, if we later on want to sort grid items (a bit far fetched example, but you get the idea).

If you still think it's better to move everything sort related into the column, I can do it.